mytoyota/models/vehicle.py

```diff
@@ -273,8 +289,51 @@ async def get_summary(
                         )
                     )
         elif summary_type == SummaryType.WEEKLY:
-            raise NotImplementedError
+            # To build a weekly summary we need to add together all the daily summaries for the
+            # week. A week starts on a Monday so data is added into build_hdc & build_summary
+            # until the current day is Sunday(i.e. last day of the week). At this point the
+            # build_hdc & build_summary are added to the response and we start again.
+            build_hdc = None
+            build_summary = None
+            start_date = None
+
+            for summary in resp.payload.summary:
+                summary.histograms.sort(
+                    key=attrgetter("day")
+                )  # Code required days are in date order
+                for histogram in summary.histograms:
+                    today = date(day=histogram.day, month=histogram.month, year=histogram.year)
```

Comment:
Why is this called `today`? 
This value is not actually today, but can change with each loop over a histogram and only shows the date present in the respective histogram, right?

Reply:
Yes its "today" in relation to the loop. I've updated the names.